Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,333 @@
+ *
+ *
+ * Example TypoScript configuration:
+ *
+ * 10 = BK2K\BootstrapPackage\DataProcessing\LanguageMenuProcessor
+ * 10 {
+ *   as = languagenavigation
+ *   expandAll = 1
+ *   includeSpacer = 1
+ *   dataProcessing {
+ *     10 = TYPO3\CMS\Frontend\DataProcessing\FilesProcessor
+ *     10 {
+ *        references.fieldName = media
+ *     }
+ *   }
+ * }
+ */
+class LanguageMenuProcessor extends MenuProcessor
+{
+    /**
+     * Gets the language data for the languageUid
+     *
+     * @param int $languageUid
+     * @return string JSON encoded data
+     */
+    protected function getLanguageData($languageUid)
+    {
+        return LanguageUtility::getLanguageData($languageUid);
+    }
+
+    /**
+     * Executes stdWrap on global properties
+     *
+     * @param array $conf
+     */
+    protected function stdWrap(&$conf)
+    {
+        if (isset($conf['languageUid.'])) {
+            $conf['languageUid'] = $this->cObj->stdWrap($conf['languageUid'], $conf['languageUid.']);
+        }
+    }
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->menuDefaults['as'] = 'languagemenu';
+        $this->menuDefaults['titleField'] = '';
+    }
+
+    /**
+     * Add loading from sys_languages in case special.value is not set
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function validateConfiguration()
+    {
+        $this->processorConfiguration = array_replace_recursive(
+            [
+                'special' => 'language',
+                'special.' => [
+                    'value.' => [
+                        'ifBlank.' => [
+                            'cObject' => 'COA',
+                            'cObject.' => [
+                                'wrap' => '0|',
+                                '10' => 'CONTENT',
+                                '10.' => [
+                                    'table' => 'sys_language',
+                                    'select.' => [
+                                        'pidInList' => 'root',
+                                        'orderBy' => 'sorting'
+                                    ],
+                                    'renderObj' => 'TEXT',
+                                    'renderObj.' => [
+                                        'wrap' => ',|',
+                                        'field' => 'uid'
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+            ],
+            $this->processorConfiguration
+        );
+
+        parent::validateConfiguration();
+    }
+
+    /**
+     * Add more language properties
+     */
+    public function prepareLevelLanguageConfiguration()
+    {
+        parent::prepareLevelLanguageConfiguration();
+
+        if ($this->menuConfig['special'] === 'language') {
```

Comment:
Menu is always a language menu, no need for checking this after making the processor standalone

Reply:
done